### ISSUE TITLE
IP Banning With Blank Duration Field

### ISSUE THREAD

**Issue by Vanndril** [myimouto/myimouto#142]
If you try to ban an IP via the blocked users page but leave the "Duration (days)" field blank, the ban goes through. On the ban page, it will say they are banned for 2017 years. In the database, however, the "expires_at" field of the "ip_bans" table will hold: "0000-00-00 00:00:00" (w/o the quotes).

When a user banned in this manner and using that IP address returns to the site, the ban is immediately removed from the list. I imagine this is because the ban ended some odd 2015 years ago. :P

In short: the "Duration (days)" field totally needs a default value. For fairly obvious reasons. XD

Centos 6.5 / PHP 5.5.12 / Nginx 1.0.15 / mysql Ver 14.14 Distrib 5.5.31
Booru Version: 1.0.6

**Comment by Parziphal** [comment]
You know it's not necessary to include your server info and booru version? hehe.

**Comment by Kille250** [comment]
thanks for the information :D

**Comment by Vanndril** [comment]
NOW you tell me? :P

I've only posted that in every bug report I've made so far. XD

**Comment by rupaheizu** [comment]
Just need a little confirmation @Parziphal, why not use unix timestamps instead of readable timestamps?

**Comment by Parziphal** [comment]
@rupaheizu you mean why do I use date-time instead of Unix timestamps? Because of reason that I don't remember, heh.

**Comment by rupaheizu** [comment]
@Parziphal, it would be easier to track and less code to write if we use unix timestamps on certain use cases.

**Comment by Parziphal** [comment]
Hey @rupaheizu, what are you exactly working on? How far are you from finishing?

**Comment by rupaheizu** [comment]
Hey @Parziphal, apologies, I've been sucked into my day job. I'm working on the ban with IPv6 compat. I'll make a PR within the week.

**Comment by Parziphal** [comment]
@rupaheizu Oh I meant, I think you were working on a new version of MyImouto or something?

**Comment by rupaheizu** [comment]
@Parziphal I'm currently writing tests for it. I'll update you guys here.

**Comment by Parziphal** [comment]
@rupaheizu So it's finished or nearly finished? What is it exactly, a whole rewrite? Is the code online somewhere to see it?

**Comment by Vanndril** [comment]
An entirely new version of myimouto? Oh my. This prospect both excites me and worries me. XD
